### PR TITLE
[tcp] remove EXCLUDE_TCPLP_LIB option

### PR DESCRIPTION
**Pull request by mariuszpos**

### CODE REVIEW

File: etc/cmake/options.cmake [hunk 1]
```diff
@@ -190,8 +190,6 @@ else()
     message(FATAL_ERROR "Invalid max RCP restoration count: ${OT_RCP_RESTORATION_MAX_COUNT}")
 endif()
 
-option(OT_EXCLUDE_TCPLP_LIB "exclude TCPlp library from build")
```
Comment (abtink) [inline etc/cmake/options.cmake:193]: Thanks @mariuszpos .

In general, make sense to me to remove this now that we have `OT_TCP`. But can removing this cause issue with projects that integrated OT stack and were using/setting this option?

Suggest either keeping it or adding checks to trigger error if we see it being used?

**Review by edmont — APPROVED** [review]

File: etc/cmake/options.cmake (hunk 1 above)
Comment (mariuszpos) [reply etc/cmake/options.cmake:193]: ok, I've added triggering such an error

**Review by abtink — APPROVED** [review]
👍

**Comment by mariuszpos** [comment]
@abtink do you know for what reason tests fail ?

**Comment by abtink** [comment]
> @abtink do you know for what reason tests fail ?
> Whether there are any dependencies to OT_EXCLUDE_TCPLP_LIB in other repositories?

I don't know. I don't think it is dependency issue with other repos. 
The builds are somehow failing.

**Review by edmont — APPROVED** [review]

File: etc/cmake/options.cmake [hunk 2]
```diff
@@ -34,6 +34,8 @@ option(OT_FTD "enable FTD" ON)
 option(OT_MTD "enable MTD" ON)
 option(OT_RCP "enable RCP" ON)
 
+option(OT_TCP "enable TCP" ON)
```
Comment (jwhui) [inline etc/cmake/options.cmake:37]: Insert this into the list of `OT_*` options below (in alphabetical order).

Comment (jwhui) [reply etc/cmake/options.cmake:36]: Do we want this to also control `OPENTHREAD_CONFIG_TCP_ENABLE`?
https://github.com/openthread/openthread/blob/6bfcc0d7d495fcc7bcf1cdd4070bee14c391ef95/src/core/config/ip6.h#L172

If so, we probably want to use `ot_option` instead.

File: etc/cmake/options.cmake [hunk 3]
```diff
@@ -33,13 +33,15 @@ option(OT_APP_RCP "enable RCP app" ON)
 option(OT_FTD "enable FTD" ON)
 option(OT_MTD "enable MTD" ON)
 option(OT_RCP "enable RCP" ON)
+option(OT_TCP "enable TCP" ON)
```
Comment (abtink) [inline etc/cmake/options.cmake:36]: We already have `OT_TCP` defined as `ot_option` in this file?

Comment (jwhui) [reply etc/cmake/options.cmake:36]: Yes, I think the issue is `OPENTHREAD_CONFIG_TCP_ENABLE` default is 1 in code, but default OFF in CMake:
https://github.com/openthread/openthread/blob/6bfcc0d7d495fcc7bcf1cdd4070bee14c391ef95/src/core/config/ip6.h#L172

We should set `OT_TCP` default `ON` in `ot_option` below instead.

File: etc/cmake/options.cmake [hunk 4]
```diff
@@ -190,8 +192,6 @@ else()
     message(FATAL_ERROR "Invalid max RCP restoration count: ${OT_RCP_RESTORATION_MAX_COUNT}")
 endif()
 
-option(OT_EXCLUDE_TCPLP_LIB "exclude TCPlp library from build")
-
```
Comment (abtink) [inline etc/cmake/options.cmake:194]: Thinking more on this, I am not sure if it is good idea to remove `OT_EXCLUDE_TCPLP_LIB` and use `OT_TCP` as the way to decide whether to include the `tcplp` library or not.

In general, we provide different ways for project integrators to configure OpenThread:
1. They can use CMake options to enable/disable features during build.
2. Or we allow then to  define the related `OPENTHREAD_COFNIG_{}` in a project/product-specific config header file (this is the model I personally recommend for products).

All CMake options have 3 states: `ON`, `OFF`, or (empty) not specified. The "not specified" state allows the second way above (using project-specific config header).


With the change in this PR, we are indirectly forcing the users to enable/disable TCP feature using `CMake` option as the only way.

File: etc/cmake/options.cmake (hunk 3 above)
Comment (abtink) [reply etc/cmake/options.cmake:36]: I am not sure about setting `OT_TCP` to be `ON` by default in Cmake.

In general, all CMake `ot_otion`s are by default set to empty (not-specified) so to allow us to fall back to `OPENTHREAD_CONFIG_{}` values (which may be provided in a project/product-specific config file).

Comment (jwhui) [reply etc/cmake/options.cmake:36]: Yeah, my point is that the reason why the build were failing before is that OpenThread core defaults to assume TCP it should support TCP. However, linking the tcplp library was disabled by default.

If we want to separately control OpenThread TCP support and linking the tcplp library, then we should probably have separate build configs.

**Review by canisLupus1313 — APPROVED** [review]

File: etc/cmake/options.cmake (hunk 4 above)
Comment (abtink) [reply etc/cmake/options.cmake:194]: I think would be good for @edmont  to review.

Removing this `if(NOT OT_EXCLUDE_TCPLP_LIB)` check always include the lib. I think  there was some situation where this was not desirable and reason for `OT_EXCLUDE_TCPLP_LIB` to be added from #7306.

Comment (edmont) [reply etc/cmake/options.cmake:194]: Hi, we have discussed this internally and this is not an issue anymore, so I'm fine with removing `OT_EXCLUDE_TCPLP_LIB`.

**Review by edmont — APPROVED** [review]

**Review by abtink — APPROVED** [review]
LGTM. Thanks.

**Review by jwhui — APPROVED** [review]

**Pull request merged**